shared/src/MessageIdDefault.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    /// <summary>
+    /// The different configurations available for setting a default value of MessageId on an IoT Hub Message.
+    /// </summary>
+    public enum MessageIdDefault
+    {
+        /// <summary>
+        /// MessageId is settable only by the user.
+        /// </summary>
+        Null,
+
+        /// <summary>
+        /// If MessageId is not set by the user, the client library will set to a random GUID before sending the message.
```

Comment:
This is really a hybrid solution then. We only set it when null.

Rather than looking at this enum as what will be set by the SDK (null isn't true), think of it as the behavior the SDK will have. Do we set it? When? To what?

Reply:
Yes, we don't want to override the value if the user already sets it to something. Only for the case when it isn't set by the user, the sdk will set it to a guid, right before sending the message.

Reply:
I didn't really get your comment @drwill-ms ; was it a question, or did you have a concern with the behavior?

Reply:
I agree with the comment, but don't think the enum names match it. It requires a comment to get across what the behavior will be. So I'm suggesting changing the enum value name from **what** the property will be set to to **when** and **what** it'll be set to.

Reply:
You've updated it and I don't have a concern here anymore.